dali/pipeline/operators/decoder/nvjpeg_decoder_cpu.h

```diff
@@ -74,10 +74,19 @@ class nvJPEGDecoderCPUStage : public Operator<CPUBackend> {
   }
 
   virtual ~nvJPEGDecoderCPUStage() noexcept(false) {
-    NVJPEG_CALL(nvjpegDecoderDestroy(decoder_host_));
-    NVJPEG_CALL(nvjpegDecoderDestroy(decoder_hybrid_));
-    NVJPEG_CALL(nvjpegDestroy(handle_));
-    PinnedAllocator::FreeBuffers();
+    try {
+      NVJPEG_CALL(nvjpegDecoderDestroy(decoder_host_));
+      NVJPEG_CALL(nvjpegDecoderDestroy(decoder_hybrid_));
+      for (int i = 0; i < batch_size_; i++) {
+        NVJPEG_CALL(nvjpegDecodeParamsDestroy(decode_params_[i]));
+      }
+      NVJPEG_CALL(nvjpegDestroy(handle_));
+      PinnedAllocator::FreeBuffers();
+    } catch (const std::exception &e) {
+      // If destroying nvJPEG resources failed we are leaking something so terminate
```

Comment:
wondering: isn't it what happens already when we throw an exception from the destructor?

Reply:
To show some reasonable message.